sdk/python/foundation-models/system/inference/text-to-image/safe-text-to-image-online-deployment.ipynb

```diff
@@ -0,0 +1,616 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# How to create an Azure AI Content Safety enabled text-to-image online endpoint (Preview)\n",
+    "### This notebook will walk you through the steps to create an __Azure AI Content Safety__ enabled __text-to-image__ online endpoint.\n",
+    "### This notebook is under preview\n",
+    "### The steps are:\n",
+    "1. Create an __Azure AI Content Safety__ resource for moderating the request from user and response from the __text-to-image__ online endpoint.\n",
+    "2. Create a new __Azure AI Content Safety__ enabled __text-to-image__ online endpoint with a custom score.py which will integrate with the __Azure AI Content Safety__ resource to moderate the response from the __text-to-image__ models and the request from the user, but to make the custom score.py to successfully authenticated to the __Azure AI Content Safety__ resource, we have 2 options:\n",
+    "    1. __UAI__, recommended but more complex approach, is to create a User Assigned Identity (UAI) and assign appropriate roles to the UAI. Then, the custom score.py can obtain the access token of the UAI from the AAD server to access the Azure AI Content Safety resource. Use [this notebook](aacs-prepare-uai.ipynb) to create UAI account for step 3 below\n",
+    "    2. __Environment variable__, simpler but less secure approach, is to just pass the access key of the Azure AI Content Safety resource to the custom score.py via environment variable, then the custom score.py can use the key directly to access the Azure AI Content Safety resource, this option is less secure than the first option, if someone in your org has access to the endpoint, he/she can get the access key from the environment variable and use it to access the Azure AI Content Safety resource.\n",
+    "  "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 1. Prerequisites\n",
+    "#### 1.1 Check List:\n",
+    "- [x] You have created a new Python virtual environment for this notebook.\n",
+    "- [x] The identity you are using to execute this notebook(yourself or your VM) need to have the __Contributor__ role on the resource group where the AML Workspace your specified is located, because this notebook will create an Azure AI Content Safety resource using that identity."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import time\n",
+    "#### 1.2 Assign variables for the workspace and deployment\n",
+    "\n",
+    "# The public registry name contains text-to-image models\n",
+    "# TODO: change registry name to azureml\n",
+    "registry_name = \"azmlft-dev-registry01\"\n",
+    "\n",
+    "# Name of the text-to-image model to be deployed\n",
+    "model_name = \"stabilityai-stable-diffusion-2-1\"\n",
```

Comment:
nit: Let's ensure to use the same model across all our notebooks.

Reply:
Thanks for catching it. Using "runwayml-stable-diffusion-v1-5" as default.